pandas/_libs/tslibs/c_timestamp.pyx

```diff
@@ -299,9 +299,15 @@ cdef class _Timestamp(datetime):
             # scalar Timestamp/datetime - Timestamp/datetime -> yields a
             # Timedelta
             from pandas._libs.tslibs.timedeltas import Timedelta
+            from pandas._libs.tslibs.timestamps import Timestamp
             try:
                 return Timedelta(self.value - other.value)
-            except (OverflowError, OutOfBoundsDatetime):
+            except (OverflowError, OutOfBoundsDatetime) as e:
+                if isinstance(other, Timestamp):
+                    raise OverflowError(
+                        "Result is too large for pandas.Timestamp. Convert inputs "
+                        "to datetime.datetime before subtracting."
```

Comment:
maybe suggest `obj.to_pydatetime()`?

I think re-raising OutOfBoundsDatetime makes sense here, as it subclasses ValueError, which is what we are catching elsewhere (my guess for why the CI is broken)